### ISSUE TITLE
space-before-function-paren fails with async keyword

### ISSUE THREAD

**Issue by fnando** [eslint/eslint#11436]
**Tell us about your environment**

* **ESLint Version:** v5.14.1
* **Node Version:** v11.6.0
* **npm Version:** 6.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```console
$ echo 'const callback = async () => {};' > space-before-function-paren.js                                            
$ eslint --no-eslintrc --parser babel-eslint --rule 'space-before-function-paren: ["error", "never"]' space-before-function-paren.js

/Users/fnando/Projects/samples/space-before-function-paren.js
  1:23  error  Unexpected space before function parentheses  space-before-function-paren

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**What did you expect to happen?**

eslint should detect the `async` keyword and allow such instruction.

**What actually happened? Please include the actual, raw output from ESLint.**

The code was detected as invalid.

**Are you willing to submit a pull request to fix this bug?**

Not sure how to do it, but sure.

**Comment by fnando** [comment]
For now, I'm disabling the rule inline:

```js
const callback = async () => { // eslint-disable-line space-before-function-paren
  // ...
};
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. This is working as intended because you have a space before your function parentheses. You should remove the space to prevent an error from being reported.

Alternatively, you could use the [`asyncArrow` option](https://eslint.org/docs/rules/space-before-function-paren) if you want to have a space for async arrow functions.

**Comment by fnando** [comment]
hey @not-an-aardvark! Ace reply time! Impressing!

And thanks for pointing me to the `asyncArrow` option. That's exactly what I needed to add.
